packages/web/src/components/search/DataSearch.js

```diff
@@ -998,21 +1023,34 @@ DataSearch.defaultProps = {
 	showClear: false,
 	strictSelection: false,
 	searchOperators: false,
+	size: 20,
 };
 
-const mapStateToProps = (state, props) => ({
-	selectedValue:
-		(state.selectedValues[props.componentId]
-			&& state.selectedValues[props.componentId].value)
-		|| null,
-	suggestions: state.hits[props.componentId] && state.hits[props.componentId].hits,
-	themePreset: state.config.themePreset,
-	isLoading: state.isLoading[props.componentId] || false,
-	error: state.error[props.componentId],
-	analytics: state.analytics,
-	config: state.config,
-	headers: state.appbaseRef.headers,
-});
+const mapStateToProps = (state, props) => {
+	const stateAggs = state.aggregations[props.componentId];
+	const computeSuggestions = () => {
```

Comment:
@ShahAnuj2610 We should not compute state in `mapStateToProps` because it can lead to performance issues. You can do this computation in `reducer` in `reactivecore`.

Reply:
https://github.com/appbaseio/reactivecore/pull/48